### PR TITLE
fix(common): make getDetailed span-consistent for WAIT-prefixed x87 instructions

### PR DESCRIPTION
**Pull request by r0ny123**
## What & why

Addresses **danielplohmann/smda#114**. `SmdaInstruction.getDetailed()` re-disassembles the stored instruction bytes with Capstone, which can split a single SMDA/IDA instruction into two. The classic case is a WAIT-prefixed x87 op:

```
9bd93c24  ->  wait                       (1 byte)
              fnstcw word ptr [esp]       (3 bytes)
```

The old code took `with_details[-1]` and logged a warning **every time**. That happened to return the right detail here, but only by luck, and it would mis-select for any non-trailing split.

## Fix

- Drop standalone `wait`/`fwait` prefix instructions, then return the trailing **operation** instruction. Its operands and its `address + size` line up with the end of the stored byte span (`0x4d3f1f1 + 3 == 0x4d3f1f0 + 4`), so the returned detail is span-consistent.
- The known WAIT-prefix pattern is now **silent**; a warning is emitted only for genuinely unexpected splits (0 or >1 non-prefix instructions), and it now names the mnemonics involved.
- Replaced the bare `assert len(...) == 1` (strippable under `python -O`) with an explicit `ValueError` for the zero-instruction case.
- Removed the stale TODO comment block.

## Tests

New `tests/testSmdaInstructionDetail.py`:
- ordinary single instruction returned unchanged
- **`9bd93c24` regression**: returns `fnstcw`, carries the memory operand, `address + size` reaches the span end, and emits **no warning**
- empty bytes raise `ValueError`
- non-Intel architecture raises `NotImplementedError`

`make test` green locally (incl. `testIntegration`, which exercises `getDetailed()` via data-ref recovery); `ruff` clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Bug Fixes**
  * Improved instruction decoding to correctly handle x87 floating-point wait instructions that may be decoded as separate operations
  * Enhanced error handling for cases where instruction bytes cannot be decoded

* **Tests**
  * Added test coverage for instruction detail extraction and related edge cases

### CODE REVIEW

File: tests/testSmdaInstructionDetail.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        self.assertEqual(detailed.size, 1)
+
+    def test_wait_prefixed_x87_returns_span_consistent_operation(self):
+        # `9bd93c24` is the WAIT-prefixed FSTCW that IDA/SMDA store as one instruction, but
+        # Capstone splits into `wait` + `fnstcw word ptr [esp]`. getDetailed() must return the
+        # x87 operation, with its memory operand and an (address + size) reaching the byte span end.
+        offset = 0x4D3F1F0
+        ins = _intel_instruction(offset, "9bd93c24", "fstcw", "word ptr [esp]")
+        with self.assertNoLogs(logger="smda.common.SmdaInstruction", level=logging.WARNING):
+            detailed = ins.getDetailed()
```
[the remark this answers is not in the document]
Comment (r0ny123) [reply tests/testSmdaInstructionDetail.py:32]: `pyproject.toml` sets `requires-python = ">=3.10"`, and `unittest.TestCase.assertNoLogs` is available from Python 3.10, so it is safe on every supported version. Leaving as-is.

**Comment by r0ny123** [comment]
**Adversarial self-review pass — cross-file regression fix (commit `4a92603`):**
Changing `getDetailed()` to raise `ValueError` (instead of the old strip-able `assert`'s `AssertionError`) was not self-contained: `IntelInstructionEscaper.escapeBinaryPtrRef()` only caught `(AttributeError, AssertionError, NotImplementedError)` to default its `pack_format`, so the new `ValueError` would have crashed escaping for an instruction with a `[disp]` operand whose bytes don't decode. Added `ValueError` to that catch + a regression test. (`NON_OPERATIONAL_EXCEPTION_TYPES` left unchanged on purpose — undecodable bytes are operational, not a programming bug.) CI green.
